nemo/collections/asr/data/audio_to_text_dataset.py

```diff
@@ -398,6 +400,88 @@ def get_tarred_dataset(
     return get_chain_dataset(datasets=datasets, ds_config=config, rank=global_rank)
 
 
+def get_code_switched_dataset(
+    config: dict,
+    shuffle_n: int,
+    global_rank: int,
+    world_size: int,
+    tokenizer: Optional['TokenizerSpec'] = None,
+    augmentor: Optional['AudioAugmentor'] = None,
+) -> CodeSwitchedDataset:
+
+    tarred_audio_filepaths = config.get('tarred_audio_filepaths', None)
+    manifest_filepaths = config['manifest_filepath']
+
+    # needed to support validation Datasets that arrive here as
+    # [[dataset1,dataset2]] otherwise ModelPT would interfere
+    if len(manifest_filepaths) == 1 and not isinstance(manifest_filepaths[0], str):
+        manifest_filepaths = config['manifest_filepath'][0]
+    if tarred_audio_filepaths is None:
+        tarred_audio_filepaths = [None] * len(manifest_filepaths)
+    datasets = {}
+    for dataset_idx, (tarred_audio_filepath, manifest_filepath) in enumerate(
+        zip(tarred_audio_filepaths, manifest_filepaths)
+    ):
+        lang = config['code_switch_languages'][dataset_idx]
```

Comment:
What happens if the a manifest contains LID information and we pass LIDs here as well?

Reply:
It's not a problem, because LIDs in the manifest always take precedence, you can see that here: https://github.com/NVIDIA/NeMo/blob/40dd53d25b0d3a40775d32c6438580cd526af2d0/nemo/collections/common/parts/preprocessing/collections.py#L160

The new language parameter is only used as a fallback IFF using AggTokeniser and the manifest has no LID in the manifest.